.circleci/config.yml

```diff
@@ -19,12 +23,13 @@ jobs:
           command: |
             python -m virtualenv env
 
-      - run: &install-dependencies
+      - run:
           name: install dependencies
           command: |
             . env/bin/activate
             python --version
             pip install -r requirements.txt -r tests/requirements.txt
+            pip install --upgrade 'dimod<< parameters.dimod-version >>'
```

Comment:
Do we want to always upgrade all dependencies as well?

For dep version sweep I [prefer](https://github.com/dwavesystems/dwave-cloud-client/blob/e5a48aff4b897d6c0a463a15d963eb587874ebeb/.circleci/config.yml#L195) to [use](https://github.com/dwavesystems/dwave-greedy/blob/aa98316140e62b81686b28384f82d3751acd2a60/.circleci/config.yml#L92) `~=a.b.c` specifier, without the `--upgrade` flag because that installs the latest, but with minimal changes to the environment (i.e. no `numpy` upgrade in this case).

@arcondello thoughts?

Reply:
No strong opinions, other than the other obvious fix (add a `--no-deps`) will stop working well when `pyparsing` is added to dimod in https://github.com/dwavesystems/dimod/pull/958